lib/league_stats.rb

```diff
@@ -46,26 +45,23 @@ def worst_offense
   def highest_scoring_visitor
     away_teams = away_games.group_by {|game| game.team_id }
     away_team_records = away_teams.transform_values do |games|
-     total_games = games.length.to_f
+     total_games = games.length
      goals = games.sum {|games| games.goals}
-     (goals / total_games)
-    end
-    highest_scoring = away_team_records.max_by do |team, average_goals|
-     average_goals
+     average(goals, total_games)
     end
+    highest_scoring = away_team_records.max_by { |team, avg_goals| avg_goals }
```

Comment:
another method that is much shorter and easier to read thanks to mathable module. nice work.